spec/intro.md

```diff
@@ -110,20 +110,19 @@ defined for _default functions_.
 > differences in implementation or changes to locale data
 > (such as due to the release of new CLDR versions).
 
-Updates to this specification will only reserve, define, or require
-_function_ _identifiers_ and _function_ _option_ _identifiers_
-which satisfy either of the following two requirements:
+**_<dfn>Reserved identifiers</dfn>_** are those that satisfy at least one of the following conditions:
+
 - Includes no _namespace_,
   and has a _name_ consisting of characters in the ranges a-z, A-Z, and 0-9,
   and the characters U+002E FULL STOP `.`, U+002D HYPHEN-MINUS `-`, and U+005F LOW LINE `_`.
 - Uses a _namespace_ consisting of a single character in the ranges a-z and A-Z.
 
-All other _identifiers_ in these categories are reserved for the use of implementations or users.
+Updates to this specification will only reserve, define, or require _reserved identifiers_.
+All _identifiers_ other than _reserved identifiers_ are available for the use of implementations or users.
 
-> [!IMPORTANT]
-> Implementation-defined or user-defined _functions_ and _function_ _options_
-> SHOULD use a _namespace_ as part of their _identifiers_
-> to help avoid collisions with other implementations.
+> [!NOTE]
+> Implementers and users are strongly advised to namespace all _identifiers_ used in custom _functions_,
+> _options_ to _default functions_, _attributes_, and _markup_.
```

Comment:
This is a normative change, since it removes a normative SHOULD (albeit to function authors and which cannot be tested). Why not keep the normative level intact? We can do a wording change, if desired.

I'm not sure about "markup", since we don't define markup.

We could make it a bit tighter:

```suggestion
> [!NOTE]
> Use a namespace for all _identifiers_ defined by an implementation
> or in the _names_ of custom _functions_, _attributes_, or _markup_ defined by users.
```

Reply:
The fuzziness of 'implementation' and 'user' gets in the way here. I read 'custom' as 'not defined by this spec. I read 'users' as 'users of an implementation', including that implementation's custom functions, etc.

Reply:
Given that the SHOULD is effectively untestable, I don't think changing it to "strongly advised" is that significant a change.

I'd be fine with leaving un-namespaced markup as explicitly not reserved.

Reply:
> I'd be fine with leaving un-namespaced markup as explicitly not reserved.

I disagree: we currently don't define any standard markup, but we could in the future (same with attributes). So the same principles for functions and attributes apply.